src/rejson_api.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include "redismodule.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum JSONType {
+    JSONType_String = 0,
+    JSONType_Int = 1,
+    JSONType_Double = 2,
+    JSONType_Bool = 3,
+    JSONType_Object = 4,
+    JSONType_Array = 5,
+    JSONType_Null = 6,
+    JSONType__EOF
+} JSONType;
+
+typedef const void *RedisJSONKey;
+typedef const void *RedisJSON;
+
+typedef struct RedisJSONAPI_V1 {
+    /* RedisJSONKey functions */
+    RedisJSONKey (*openKey)(RedisModuleCtx *ctx, RedisModuleString *key_name);
+    RedisJSONKey (*openKeyFromStr)(RedisModuleCtx *ctx, const char *path);
+
+    void (*closeKey)(RedisJSONKey key);
+
+    /* RedisJSON functions
+     * Return NULL if path does not exist
+     * `count` can be NULL and return 0 for non array/object
+     **/
+    RedisJSON (*get)(RedisJSONKey key, const char *path, JSONType *type, size_t *count);
+
+    RedisJSON (*getAt)(RedisJSON jsonIn, size_t index, JSONType *type, size_t *count);
```

Comment:
I guess this API only relevant for array? what happened if used on none array json?
Same questions and comment here about type and count

Reply:
Same comment here about the count and type

Reply:
@MeirShpilraien @ashtul If not an Object or Array  (Object also has iterator `nth`) or if index is out of range, `NULL` is returned (we do **not** propagate the error message to the user, e.g., `"JSON index is out of range"` or `"Not a JSON Array or Object"`)

Reply:
As commented above, I believe count and type should be retrieve by a seperate API on RedisJSON object.